### DISCUSSION TITLE
How to implement batch nft minting transactions?

### DISCUSSION THREAD

**Discussion by JohnVersus** [metaplex-foundation/js#362] [Q&A]
Hi @lorisleiva , 
My question is related to issue #19. Is there an example of how to use this with Metaplex transactions?

I want to batch mint nft's using `metaplex.nfts().create` without having to sign multiple transactions. Is it possible to achieve this by using `signAllTransactions` or through any other way?

**Comment by gabspeck** [comment]
Hi @JohnVersus , in case you're still struggling with this, I had the same requirement in my project and below is a rough sketch of how I implemented it. My project is still on version 0.16.1 of the SDK, I think the syntax may have changed a little since then:

```javascript
import {Metaplex, walletAdapterIdentity} from "@metaplex-foundation/js";
import {clusterApiUrl, Keypair} from "@solana/web3.js";

const BATCH_SIZE = 10
const metaplex = new Metaplex(new Connection(clusterApiUrl('devnet')))
const sourceArray = []
for (let i = 0; i < BATCH_SIZE; i++) {
    sourceArray.push(i)
}
const updateAuthority = Keypair.generate()
const recentBlockhash = await metaplex.rpc().getLatestBlockhash()

const txs = await Promise.all(sourceArray.map(async i => {
    const useNewMint = Keypair.generate()

    const tx = (await metaplex
        .use(walletAdapterIdentity(wallet))
        .nfts()
        .builders()
        .create({
            useNewMint,
            updateAuthority,
            uri: 'http://example.com/mynft',
            sellerFeeBasisPoints: 1000,
            name: `NFT #${i}`
        })).toTransaction()
    tx.recentBlockhash = recentBlockhash
    tx.partialSign(useNewMint, updateAuthority)
    return tx
}))

const signedTxs = await wallet.signAllTransactions(txs)
const hashes = await Promise.all(signedTxs.map(tx => metaplex.connection.sendRawTransaction(tx)))
await Promise.all(hashes.map(h => metaplex.rpc().confirmTransaction(h)))
```

PS: I haven't actually run this code, there might be an error or another, but roughly speaking this is what worked for me.

**Reply by JohnVersus** [reply]
Thank Bro. I learned something new with this. 
This logic still wants me sign multiple transactions, which could be due to the error which I am seeing. I will look more into it.
<img width="716" alt="image" src="https://user-images.githubusercontent.com/15834299/197930269-ef510778-a13f-4d64-accf-53f159bd962c.png">

**Reply by shantanuSakpal** [reply]
any luck?
